File: Programming Basics with Python/While-цикъл/graduation.py

name = input()

passing_grade = 4.0
sum_garde = 0.0
average_grade = 0.0

count_strikes = 0
count_grade = 0

while True:
    grade = float(input())
    count_grade += 1
    
    if grade < passing_grade:
        count_strikes += 1
        if count_strikes == 2:
            count_grade -= 1
            print(f"{name} has been excluded at {count_grade} grade")
            break
    else:
        sum_garde += grade
    
    if count_grade == 12:
        average_grade = sum_garde / count_grade
        print(f"{name} graduated. Average grade: {average_grade:.2f}")
        break
    

File: Programming Basics with Java/While-loop/Graduation.java
import java.util.Arrays;
import java.util.Scanner;

public class Graduation 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String name = input.nextLine();

        double[] arr = {};
        double sum = 0.0;
        int classes = 0;
        int mark = 0;

        while (classes != 12)
        {
            classes += 1;
            double grade = Double.parseDouble(input.nextLine());

            arr = Arrays.copyOf(arr, arr.length + 1);
            arr[arr.length - 1] = grade;

            if (grade < 4)
            {
                mark += 1;
                if (mark >= 2)
                {
                    System.out.printf("%s has been excluded at %d grade", name, arr.length - 1);
                    break;
                }
            }

            if (classes == 12)
            {
                for (int i = 0; i < arr.length; i++)
                {
                    sum += arr[i];
                }

                double avg = sum / arr.length;
                System.out.printf("%s graduated. Average grade: %.2f", name, avg);
                break;
            }
        }
    }
}
